src/layout/page-questionnaire/components/page-questionnaire.jsx

```diff
@@ -40,99 +40,97 @@ export const defaultProps = {
   collectedVariablesByQuestion: {},
 };
 
-class PageQuestionnaire extends Component {
-  static propTypes = propTypes;
-  static defaultProps = defaultProps;
-
-  UNSAFE_componentWillMount() {
-    const {
-      id,
-      questionnaire,
-      components,
-      codeLists,
-      calculatedVariables,
-      externalVariables,
-      collectedVariablesByQuestion,
-    } = this.props;
-
-    this.props.loadQuestionnaireIfNeeded(id);
-
-    if (questionnaire.id) {
-      const idCampaign = questionnaire.campaigns[0];
-      this.props.setActiveQuestionnaire(questionnaire);
-      this.props.setActiveComponents(components);
-      this.props.setActiveCodeLists(codeLists);
-      this.props.setActiveVariables({
-        activeCalculatedVariablesById: calculatedVariables,
-        activeExternalVariablesById: externalVariables,
-        collectedVariableByQuestion: collectedVariablesByQuestion,
+const PageQuestionnaire = props => {
+  const {
+    id,
+    questionnaire,
+    components,
+    codeLists,
+    calculatedVariables,
+    externalVariables,
+    collectedVariablesByQuestion,
+    activeQuestionnaire,
+  } = props;
+  
+  const [idState, setIdState] = useState(id);
+  const [questionnaireState, setQuestionnaireState] = useState(questionnaire);
+  const [activeQuestionnaireState, setActiveQuestionnaireState] = useState(activeQuestionnaire);
+  const [componentsState, setComponentsState] = useState(components);
+  const [codeListsState, setCodeListsState] = useState(codeLists);
+  const [externalVariablesState, setExternalVariables] = useState(externalVariables);
+  const [calculatedVariablesState, setCalculatedVariables] = useState(calculatedVariables);
+  const [collectedVariablesByQuestionState, setCollectedVariablesByQuestion] = useState(collectedVariablesByQuestion);
+
+  useEffect(() => {
```

Comment:
You could maybe create multiple useEffect instead of having one big useEffect. What do you think ?